tests/utils/test_fmrisim.py

```diff
@@ -193,36 +362,167 @@ def test_generate_noise():
                               )
 
     # Generate the mask of the signal
-    mask, template = sim.mask_brain(signal, mask_threshold=0.1)
+    mask, template = sim.mask_brain(signal,
+                                    mask_self=None)
 
     assert min(mask[mask > 0]) > 0.1, "Mask thresholding did not work"
     assert len(np.unique(template) > 2), "Template creation did not work"
 
     stimfunction_tr = stimfunction[::int(tr_duration * 100)]
+
     # Create the noise volumes (using the default parameters)
     noise = sim.generate_noise(dimensions=dimensions,
                                stimfunction_tr=stimfunction_tr,
                                tr_duration=tr_duration,
                                template=template,
                                mask=mask,
+                               iterations=[1, 0],
                                )
 
     assert signal.shape == noise.shape, "The dimensions of signal and noise " \
                                         "the same"
 
-    assert np.std(signal) < np.std(noise), "Noise was not created"
-
-    noise = sim.generate_noise(dimensions=dimensions,
-                               stimfunction_tr=stimfunction_tr,
-                               tr_duration=tr_duration,
-                               template=template,
-                               mask=mask,
-                               noise_dict={'sfnr': 10000, 'snr': 10000},
-                               )
-
-    system_noise = np.std(noise[mask > 0], 1).mean()
+    noise_high = sim.generate_noise(dimensions=dimensions,
+                                    stimfunction_tr=stimfunction_tr,
+                                    tr_duration=tr_duration,
+                                    template=template,
+                                    mask=mask,
+                                    noise_dict={'sfnr': 50, 'snr': 25},
+                                    iterations=[1, 0],
+                                    )
+
+    noise_low = sim.generate_noise(dimensions=dimensions,
+                                   stimfunction_tr=stimfunction_tr,
+                                   tr_duration=tr_duration,
+                                   template=template,
+                                   mask=mask,
+                                   noise_dict={'sfnr': 100, 'snr': 25},
+                                   iterations=[1, 0],
+                                   )
+
+    system_high = np.std(noise_high[mask > 0], 1).mean()
+    system_low = np.std(noise_low[mask > 0], 1).mean()
+
+    assert system_low < system_high, "SFNR noise could not be manipulated"
```

Comment:
I am slightly confused here. If SFNR is larger, does it mean that mean intensity is higher with the same amount of noise? Does that mean noise_low should have a higher average value than noise_high (given that sfnr is larger in noise_low)?